File: python/6kyu - H.py
# Link -> https://www.codewars.com/kata/54b724efac3d5402db00065e

def decodeMorse(morse_code):
    morse_dic = {
        '.-': 'A',
        '-...': 'B',
        '-.-.': 'C',
        '-..': 'D',
        '.': 'E',
        '..-.': 'F',
        '--.': 'G',
        '....': 'H',
        '..': 'I',
        '.---': 'J',
        '-.-': 'K',
        '.-..': 'L',
        '--': 'M',
        '-.': 'N',
        '---': 'O',
        '.--.': 'P',
        '--.-': 'Q',
        '.-.': 'R',
        '...': 'S',
        '-': 'T',
        '..-': 'U',
        '...-': 'V',
        '.--': 'W',
        '-..-': 'X',
        '-.--': 'Y',
        '--..': 'Z',
        '-----': '0',
        '.----': '1',
        '..---': '2',
        '...--': '3',
        '....-': '4',
        '.....': '5',
        '-....': '6',
        '--...': '7',
        '---..': '8',
        '----.': '9',
        '...---...': 'SOS',
        '-.-.--': '!',
        '.-.-.-': '.'
    }

    message = ''
    words = morse_code.strip().split('   ')

    for i in words:
        letters = i.split(' ')

        for j in letters:
            message += morse_dic[j]

        message += ' '

    return message.rstrip()

File: C#/6kyu - H.cs
//Link -> https://www.codewars.com/kata/5544c7a5cb454edb3c000047

public class BouncingBall {
  
  public static int bouncingBall(double h, double bounce, double window) {
      // your code
       int sights = 1;

            if(h > 0 && bounce < 1 && 0 < bounce && h > window)
            {
                do
                {
                    if (h * bounce > window)
                    {
                        sights += 2;
                    }

                    h = h * bounce;
                }
                while (h > window);

                return sights;
            }
            else
            {
                return -1;
            }
  }
}
